sim/midas/src/main/cc/simif.h

```diff
@@ -21,6 +25,41 @@ double diff_secs(midas_time_t end, midas_time_t start);
 typedef std::map<std::string, size_t> idmap_t;
 typedef std::map<std::string, size_t>::const_iterator idmap_it_t;
 
+/**
+ * Interface for a simulation implementation.
+ *
+ * `simif_t` interacts with the simulation, initialising it, running and and
+ * running finalisation logic at the end.  All simulation implementations
+ * (midasexamples, firesim, bridges, fasedtests) derive this interface,
+ * implementing the driver-specific logic.
+ */
+class simulation_t {
```

Comment:
So, on some level this is a very appropriate name. However, it's also highly overloaded in our context. You say
`driver-specific logic`, so maybe "Driver" or "driver_t"? or "user_driver"? I don't know.

Reply:
Yes, I think `driver_t` might be more appropriate. But I'll leave it for a later PR.